validator/node/node.go

```diff
@@ -254,81 +242,34 @@ func (c *ValidatorClient) initializeFromCLI(cliCtx *cli.Context, router *http.Se
 		}
 	}
 
-	if err := c.initializeDB(cliCtx); err != nil {
+	if err := c.initializeDB(); err != nil {
 		return errors.Wrapf(err, "could not initialize database")
 	}
 
-	if !cliCtx.Bool(cmd.DisableMonitoringFlag.Name) {
-		if err := c.registerPrometheusService(cliCtx); err != nil {
-			return err
-		}
-	}
-	if err := c.registerValidatorService(cliCtx); err != nil {
-		return err
-	}
-	if cliCtx.Bool(flags.EnableRPCFlag.Name) {
-		if err := c.registerRPCService(router); err != nil {
-			return err
-		}
+	if err := c.registerPrometheusService(); err != nil {
+		return errors.Wrapf(err, "could not register prometheus service")
 	}
-	return nil
-}
-
-func (c *ValidatorClient) initializeForWeb(cliCtx *cli.Context, router *http.ServeMux) error {
-	if cliCtx.IsSet(flags.Web3SignerURLFlag.Name) {
-		// Custom Check For Web3Signer
-		c.wallet = wallet.NewWalletForWeb3Signer(cliCtx)
-	} else {
-		// Read the wallet password file from the cli context.
-		if err := setWalletPasswordFilePath(cliCtx); err != nil {
-			return errors.Wrap(err, "could not read wallet password file")
-		}
 
-		// Read the wallet from the specified path.
-		w, err := wallet.OpenWalletOrElseCli(cliCtx, func(cliCtx *cli.Context) (*wallet.Wallet, error) {
-			return nil, nil
-		})
-		if err != nil {
-			return errors.Wrap(err, "could not open wallet")
-		}
-		c.wallet = w
+	if err := c.registerValidatorService(); err != nil {
+		return errors.Wrapf(err, "could not register validator service")
 	}
 
-	if err := c.initializeDB(cliCtx); err != nil {
-		return errors.Wrapf(err, "could not initialize database")
+	if err := c.registerRPCService(); err != nil {
+		return errors.Wrapf(err, "could not register RPC service")
 	}
 
-	if !cliCtx.Bool(cmd.DisableMonitoringFlag.Name) {
-		if err := c.registerPrometheusService(cliCtx); err != nil {
-			return err
-		}
-	}
-	if err := c.registerValidatorService(cliCtx); err != nil {
-		return err
-	}
-
-	if err := c.registerRPCService(router); err != nil {
-		return err
-	}
-
-	host := cliCtx.String(flags.HTTPServerHost.Name)
-	port := cliCtx.Int(flags.HTTPServerPort.Name)
-	webAddress := fmt.Sprintf("http://%s:%d", host, port)
-	log.WithField("address", webAddress).Info(
-		"Starting Prysm web UI on address, open in browser to access",
-	)
 	return nil
 }
 
-func (c *ValidatorClient) initializeDB(cliCtx *cli.Context) error {
```

Comment:
What is the reasoning for moving from passing the cliCtx as an argument to accessing it a struct field on the method receiver? I would like for us to move cliCtx away from runtime code. The end state would be to process flags outside of a component like ValidatorClient - translate those flags into the options exposed by the components themselves, similar to how we are trending in the beacon-chain cmd. This decouples the components from the flag system. The changes in this PR, to use a cliCtx struct field instead of passing in the cliCtx, go in the opposite direction.

Reply:
I see we wanted to do more functional options but that's not part of this or. I just noticed it was set in the validator so I thought I could reuse it. I can revert this part of the change and remove the clictx from the validator struct

Reply:
I do think leave the cliCtx plumbing alone would keep the PR focused on the main goals.